app/controllers/application_controller.rb

```diff
@@ -1,2 +1,30 @@
 class ApplicationController < ActionController::API
+  private
+
+  set_current_tenant_through_filter
+  before_action :set_the_current_tenant
+
+  def set_the_current_tenant
+    return unless ENV["ENFORCE_TENANCY"]
+
+    tenant = Tenant.find_or_create_by(:external_tenant => user_identity) if user_identity
+    if tenant
+      set_current_tenant(tenant)
+    else
+      render :json => { :errors => "Unauthorized" }, :status => :unauthorized
+    end
+  end
+
+  def user_account_number
+    Base64.decode64(request.headers.fetch_path("x-rh-identity", "identity", "account_number"))
+  end
+
+  def user_identity
```

Comment:
The method name `user_identity` is misleading.  Something like `user_account_number` or `identity_account_number` would be more accurate.